README.md

```diff
@@ -9,7 +8,186 @@ NestJS-Matic
+
+```ts
+import { MaticModule } from 'nestjs-matic';
+
+@Module({
+  imports: [MaticModule.forRoot(MATIC_OPTIONS)],
+  ...
+})
+class MyModule {}
+```
+
+**NOTE:** we describe the [MATIC_OPTIONS](#configuration-params) down below.
+
+### Configuration params
+
+The options object (`MATIC_OPTIONS`) for `PoS` and `Plasma` Client has the same composition (according to the [Matic Official PoS Doc](https://maticnetwork.github.io/matic.js/docs/pos/initialize/#options) and [Matic Official Plasma Doc](https://maticnetwork.github.io/matic.js/docs/plasma/initialize#options)). _Your matic options object must have the following structure:_
+
+```ts
+interface MaticModuleOptions {
+  /**
+   * Optional parameter for connection, must be a MaticNetworks enum.
+   * Take 'mainnet' or 'testnet' as possible values.
+   * If no network is provided, mainnnet is used.
+   * @see {@link https://maticnetwork.github.io/matic.js/docs/plasma/initialize#options}
+   */
+  network?: MaticNetworks;
+
+  /**
+   * Optional parameter for version, must be a MaticVersions enum.
+   * Take 'v1' or 'mumbai' as possible values.
+   * If no version is provided, v1 is used.
+   * @see {@link https://maticnetwork.github.io/matic.js/docs/plasma/initialize#options}
+   */
+  version?: MaticVersions;
+
+  /**
+   * Optional parameter for maticProvider, can be a string, a
+   * HttpProvider object or a WalletConnectProvider object.
+   * If no maticProvider is provided, empty object is used
+   * The maticProvider may also be a URL to connect to,
+   * such as https://rpc-mumbai.matic.today
+   * @see {@link https://github.com/ChainSafe/web3.js/tree/1.x/packages/web3-core}
+   * @see {@link https://www.npmjs.com/package/@walletconnect/ethereum-provider}
+   */
+  maticProvider?: string | HttpProvider | WalletConnectProvider;
+
+  /**
+   * Optional parameter for parentProvider, can be a string, a
+   * HttpProvider object or a WalletConnectProvider object.
+   * If no parentProvider is provided, empty object is used.
+   * The parentProvider may also be a URL to connect to,
+   * such as https://rpc.goerli.mudit.blog
+   * @see {@link https://github.com/ChainSafe/web3.js/tree/1.x/packages/web3-core}
+   * @see {@link https://www.npmjs.com/package/@walletconnect/ethereum-provider}
+   */
+  parentProvider?: string | HttpProvider | WalletConnectProvider;
+
+  /**
+   * Optional parameter for parentDefaultOptions, must be a Matic
+   * SendOptions type.
+   * If no parentDefaultOptions is provided, empty object is used.
+   * @see {@link https://github.com/maticnetwork/matic.js/blob/241d83873bb02501dd01243cd43e31265bc6ba0c/src/types/Common.ts#L3}
+   */
+  parentDefaultOptions?: SendOptions;
+
+  /**
+   * Optional parameter for maticDefaultOptions, must be a Matic
+   * SendOptions type.
+   * If no parentDefaultOptions is provided, empty object is used.
+   * @see {@link https://github.com/maticnetwork/matic.js/blob/241d83873bb02501dd01243cd43e31265bc6ba0c/src/types/Common.ts#L3}
```

Comment:
Use the link to the master branch https://github.com/maticnetwork/matic.js/blob/master/src/types/Common.ts#L3